internal/firebase/client.go

```diff
@@ -0,0 +1,123 @@
+package firebase
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-version"
+)
+
+const (
+	CloudQueryRegistryURL              = "https://firestore.googleapis.com/v1/projects/hub-cloudquery/databases/(default)/documents/orgs/"
+	CloudQueryRegistryURLWithProviders = CloudQueryRegistryURL + "%s/providers/%s"
+	providersVersionsPath              = "%s/providers/%s/versions"
+	policiesVersionPath                = "%s/policies/%s/versions"
+)
+
+type FirebaseClient struct {
+	url string
+}
+
+func NewFirebaseClient(registryUrl string) *FirebaseClient {
```

Comment:
```suggestion
func New(registryUrl string) *Client {
```

Reply:
Thanks for the review! Good point, done in https://github.com/cloudquery/cloudquery/pull/618/commits/1c8f7762d50bcbd1e3e6b291b7e1fd298189e8b9

Reply:
Missed the type rename though 😂

Reply:
Ah, good catch. Done in https://github.com/cloudquery/cloudquery/pull/622